html/janus.js

```diff
@@ -700,6 +700,11 @@ function Janus(gatewayCallbacks) {
 				// Send to generic callback (?)
 				Janus.debug("No provided notification callback");
 			}
+		} else if(json["janus"] === "timeout") {
+		    Janus.error("Timeout on session " + sessionId);
+		    Janus.debug(json);
+		    ws.close(3504, "Gateway timeout");
```

Comment:
This seems to assume this callback will always be triggered by a websocket, which is not the case. The same callback is used for HTTP as well, which would break here.

Reply:
Good point :+1: will fix it